From 07cce577e32beca853eac125917d43dffbeb534f Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 26 Jun 2019 10:26:57 +0200
Subject: [PATCH 310/386] arm64: dts: cn913x-D: add new topology with internal
 eth mode

Add support for configuration D, which is the same as
"cn913x-db-A" but with internal phy-mode configuration
set for CP0 PPv2 port0.

Change-Id: Ib6c843fb1f5766fcdd757298d96305f347efec5d
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/11262
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/cn913x.txt           | 34 +++------
 arch/arm64/boot/dts/marvell/Makefile          |  3 +
 arch/arm64/boot/dts/marvell/cn9130-db-D.dts   | 20 +++++
 arch/arm64/boot/dts/marvell/cn9131-db-D.dts   | 29 +++++++
 arch/arm64/boot/dts/marvell/cn9132-db-D.dts   | 76 +++++++++++++++++++
 5 files changed, 138 insertions(+), 24 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/cn9130-db-D.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn9131-db-D.dts
 create mode 100644 arch/arm64/boot/dts/marvell/cn9132-db-D.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
index 97eaa754e9e5..c87bfc1efb21 100644
--- a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
@@ -11,35 +11,21 @@ SoC:
       when the SoC being used is the CN9130
 
 Boards:
-   - "marvell,cn9130-db-A"
+   - "marvell,cn9130-db-A", "marvell,cn9130-db-B", "marvell,cn9130-db-D"
       when the board being used is the OcteonTX2 CN9130 Development board with
-      Setup A.
+      Setup A, B or D.
 
-   - "marvell,cn9130-db-B"
-      when the board being used is the OcteonTX2 CN9130 Development board with
-      Setup B.
-
-   - "marvell,cn9130-crb-A"
-      when the board being used is the OcteonTX2 CN9130 crb board with Setup A.
-
-   - "marvell,cn9130-crb-B"
-      when the board being used is the OcteonTX2 CN9130 crb board with Setup B.
+   - "marvell,cn9130-crb-A", "marvell,cn9130-crb-B"
+      when the board being used is the OcteonTX2 CN9130 crb board with Setup A
+      or B.
 
-   - "marvell,cn9131-db-A"
+   - "marvell,cn9131-db-A", "marvell,cn9131-db-B", "marvell,cn9131-db-D"
       when the board being used is the OcteonTX2 CN9131 Development board with
-      Setup A.
-
-   - "marvell,cn9131-db-B"
-      when the board being used is the OcteonTX2 CN9131 Development board with
-      Setup B.
-
-   - "marvell,cn9132-db-A", "marvell,cn9132-db"
-      when the board being used is the OcteonTX2 CN9132 Development board with
-      Setup A.
+      Setup A, B or D.
 
-   - "marvell,cn9132-db-B"
-      when the board being used is the OcteonTX2 CN9132 Development board with
-      Setup B.
+   - "marvell,cn9132-db-A", "marvell,cn9132-db-B", "marvell,cn9132-db-D",
+     "marvell,cn9132-db" when the board being used is the OcteonTX2 CN9132
+     Development board with Setup A, B or D.
 
 Example:
 
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 110e3bb65d19..57f3b5b65cc7 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -27,9 +27,12 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-crb-B.dtb
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-D.dts b/arch/arm64/boot/dts/marvell/cn9130-db-D.dts
new file mode 100644
index 000000000000..53e36333a515
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-D.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2019 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell CN9130 Development board. This file supports
+ * option D configuration, which is the same as "cn9130-db-A.dts" but with
+ * internal phy-mode configuration.
+ */
+
+#include "cn9130-db-A.dts"
+
+/ {
+	model = "Model: Marvell CN9130 development board (CP NOR) setup(D)";
+	compatible = "marvell,cn9130-db-D", "marvell,armada-ap807-quad",
+		     "marvell,armada-ap807";
+};
+
+&cp0_eth0 {
+	phy-mode = "internal";
+};
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db-D.dts b/arch/arm64/boot/dts/marvell/cn9131-db-D.dts
new file mode 100644
index 000000000000..0a1dc80e5bcf
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9131-db-D.dts
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:    GPL-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "cn9130-db-D.dts"
+#include "cn9131-db.dtsi"
+
+/ {
+	model = "Marvell CN9131 development board (CP NOR) setup(D)";
+	compatible = "marvell,cn9131-db-D", "marvell,armada-ap807-quad",
+		     "marvell,armada-ap807";
+};
+
+&cp1_ethernet {
+	status = "okay";
+};
+
+/* CON50 */
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp1_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&cp1_sfp_eth1>;
+};
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db-D.dts b/arch/arm64/boot/dts/marvell/cn9132-db-D.dts
new file mode 100644
index 000000000000..4af13e149cb7
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9132-db-D.dts
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:    GPL-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "cn9131-db-D.dts"
+#include "cn9132-db.dtsi"
+
+/ {
+	model = "Marvell CN9132 development board (CP NOR) setup(D)";
+	compatible = "marvell,cn9132-db-D", "marvell,armada-ap807-quad",
+		     "marvell,armada-ap807";
+
+	aliases {
+		gpio5 = &cp2_gpio1;
+		gpio6 = &cp2_gpio2;
+		ethernet5 = &cp2_eth0;
+	};
+};
+
+&cp2_ethernet {
+	status = "okay";
+};
+
+/* SLM-1521-V2, CON9 */
+&cp2_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&cp2_sfp_eth0>;
+};
+
+/* SLM-1521-V2, CON6 */
+&cp2_pcie0 {
+	status = "okay";
+	num-lanes = <2>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy0 0
+		&cp2_comphy1 0>;
+};
+
+/* SLM-1521-V2, CON8 */
+&cp2_pcie2 {
+	status = "okay";
+	num-lanes = <1>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy5 2>;
+};
+
+&cp2_sata0 {
+	status = "okay";
+	/* SLM-1521-V2, CON4 */
+	sata-port@0 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp2_comphy2 0>;
+	};
+};
+
+/* CON 2 on SLM-1683 - microSD */
+&cp2_sdhci0 {
+	status = "okay";
+};
+
+/* SLM-1521-V2, CON11 */
+&cp2_usb3_1 {
+	status = "okay";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp2_comphy3 1>;
+};
-- 
2.17.1

